From 4b263a4e1e1302f3ce0412cd12d63917e1a7123d Mon Sep 17 00:00:00 2001
From: Stefan Szabo <szbs001@freescale.com>
Date: Thu, 17 May 2012 14:21:49 +0000
Subject: [PATCH 060/121] powerpc/corenet: 14 Gb/s configuration

New default FMan configuration (14G/FMan):
* Enable out-of-the-box the automatic FMan resource (FIFOs, TNUMs, DMAs)
  allocation configuration,
* disable one 1Gb Ethernet port of each FMan on P3/5 platforms
  (one that will -- in FSL-shipped configurations -- likely be NC,
   or connected to an SGMII),
* disable three of the Offline Parsing ports (save for O/H port #0,
  which is required for host commands),
This is the maximal configuration that supports jumbo frames out of the box
(with jumbo NOT enabled out-of-the-box).

Signed-off-by: Stefan Szabo <szbs001@freescale.com>
[Kevin: Original patch taken from FSL
QorIQ-SDK-V1.2-SOURCE-20120614-yocto.iso image. Remove the mods for
 specific board's dts and default configuration files.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
Integrated-by: Yongli He <yongli.he@windriver.com>
---
 arch/powerpc/boot/dts/fsl/qoriq-fman-0.dtsi |    3 +++
 arch/powerpc/boot/dts/fsl/qoriq-fman-1.dtsi |    3 +++
 arch/powerpc/boot/dts/p5020ds.dts           |    3 +++
 arch/powerpc/boot/dts/p5040ds.dts           |    2 ++
 4 files changed, 11 insertions(+)

diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman-0.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman-0.dtsi
index 0eb551e..b074d13 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman-0.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman-0.dtsi
@@ -91,18 +91,21 @@ fman0: fman@400000 {
 		cell-index = <4>;
 		compatible = "fsl,fman-port-oh";
 		reg = <0x85000 0x1000>;
+		status = "disabled";
 	};
 
 	fman0_oh5: port@86000 {
 		cell-index = <5>;
 		compatible = "fsl,fman-port-oh";
 		reg = <0x86000 0x1000>;
+		status = "disabled";
 	};
 
 	fman0_oh6: port@87000 {
 		cell-index = <6>;
 		compatible = "fsl,fman-port-oh";
 		reg = <0x87000 0x1000>;
+		status = "disabled";
 	};
 
 	policer@c0000 {
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman-1.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman-1.dtsi
index c2c241c..d94f6cc 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman-1.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman-1.dtsi
@@ -91,18 +91,21 @@ fman1: fman@500000 {
 		cell-index = <4>;
 		compatible = "fsl,fman-port-oh";
 		reg = <0x85000 0x1000>;
+		status = "disabled";
 	};
 
 	fman1_oh5: port@86000 {
 		cell-index = <5>;
 		compatible = "fsl,fman-port-oh";
 		reg = <0x86000 0x1000>;
+		status = "disabled";
 	};
 
 	fman1_oh6: port@87000 {
 		cell-index = <6>;
 		compatible = "fsl,fman-port-oh";
 		reg = <0x87000 0x1000>;
+		status = "disabled";
 	};
 
 	policer@c0000 {
diff --git a/arch/powerpc/boot/dts/p5020ds.dts b/arch/powerpc/boot/dts/p5020ds.dts
index e775def..6bc361d 100644
--- a/arch/powerpc/boot/dts/p5020ds.dts
+++ b/arch/powerpc/boot/dts/p5020ds.dts
@@ -260,6 +260,7 @@
 					#size-cells = <0>;
 					compatible = "fsl,hydra-mdio";
 					fsl,mdio-handle = <&mdio0>;
+					fsl,hydra-mdio-mux-mask = <0x70>;
 					fsl,hydra-mdio-muxval = <0x00>;
 					status = "disabled";
 
@@ -285,6 +286,7 @@
 					#size-cells = <0>;
 					compatible = "fsl,hydra-mdio";
 					fsl,mdio-handle = <&mdio0>;
+					fsl,hydra-mdio-mux-mask = <0x70>;
 					fsl,hydra-mdio-muxval = <0x00>;
 					status = "disabled";
 
@@ -539,6 +541,7 @@
 			compatible = "fsl,p5020-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,qman-channel = <&qpool1>;
 			fsl,fman-mac = <&enet2>;
+			status = "disabled";
 		};
 		ethernet@3 {
 			compatible = "fsl,p5020-dpa-ethernet", "fsl,dpa-ethernet";
diff --git a/arch/powerpc/boot/dts/p5040ds.dts b/arch/powerpc/boot/dts/p5040ds.dts
index 9bee35b..58f02c6 100644
--- a/arch/powerpc/boot/dts/p5040ds.dts
+++ b/arch/powerpc/boot/dts/p5040ds.dts
@@ -561,6 +561,7 @@
 		ethernet@0 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&fm1dtsec1>;
+			status = "disabled";
 		};
 		ethernet@1 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
@@ -585,6 +586,7 @@
 		ethernet@6 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&fm2dtsec1>;
+			status = "disabled";
 		};
 		ethernet@7 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
-- 
1.7.9.7

